assembl/tasks/deploy.py

```diff
@@ -0,0 +1,583 @@
+def setup_aws_default_region(c):
+    assert running_locally(c)
+    region = get_and_set_region(c)
+    assert region
+    c.run('aws configure set region ' + region)
+
+
+CELERY_YAML = """_extends: cicd.yaml
+supervisor:
+  autostart_celery: true
+  autostart_celery_notify_beat: true
+  autostart_uwsgi: false
+"""
+
+
+@task()
+def get_aws_invoke_yaml(c, celery=False):
+    assert running_locally(c)
+    account = get_aws_account_id(c)
+    assert account, "Could not get aws account"
+    # This introduces a convention: yaml files
+    # for a given amazon account will be stored in
+    # s3://assembl-data-{account_id}/{fname}.yaml
+    bucket = 'assembl-data-' + account
+    invoke_path = os.path.join(c.config.projectpath, 'invoke.yaml')
+    if celery:
+        with open(invoke_path, 'w') as f:
+            f.write(CELERY_YAML)
+        invoke_path = os.path.join(c.config.projectpath, 'cicd.yaml')
+    content = get_s3_file(
+        bucket,
+        'cicd.yaml',
+        invoke_path)
+    if not content:
+        content = '_extends: terraform.yaml\n'
+        with open(invoke_path, 'w') as f:
+            f.write(content)
+    extends_re = re.compile(r'\b_extends:\s*(\w+\.yaml)')
+    match = extends_re.search(content)
+    while match:
+        key = match.group(1)
+        # assuming local bucket; what if from shared region?
+        # maybe look for client_id in bucket name, assume shared otherwise???
+        ex_content = get_s3_file(bucket, key)
+        if not ex_content:
+            break
+        with open(os.path.join(c.config.projectpath, key), 'w') as f:
+            f.write(ex_content)
+        match = extends_re.search(ex_content)
+    if not content:
+        raise RuntimeError("invoke.yaml was not defined in S3" % account)
+
+
+@task()
+def ensure_aws_invoke_yaml(c):
+    if not exists(c, 'invoke.yaml'):
+        get_aws_invoke_yaml(c)
+
+
+def is_supervisord_running(c):
+    with venv(c, True):
+        result = c.run('supervisorctl pid')
+    if not result or 'no such file' in result.stdout or 'refused connection' in result.stdout:
+        return False
+    try:
+        pid = int(result.stdout)
+        if pid:
+            return True
+    except ValueError:
+        return False
+
+
+@task()
+def supervisor_shutdown(c):
+    """
+    Shutdown Assembl's supervisor
+    """
+    if not is_supervisord_running(c):
+        return
+    with venv(c, True):
+        c.run("supervisorctl shutdown")
+    for i in range(10):
+        sleep(6)
+        with venv(c, True):
+            result = c.run("supervisorctl status", warn=True)
+        if not result.failed:
+            break
+    # Another supervisor, upstart, etc may be watching it, give it more time
+    sleep(6)
+
+
+@task()
+def supervisor_restart(c):
+    """Restart supervisor itself."""
+    supervisor_shutdown(c)
+    with venv(c, True):
+        result = c.run("supervisorctl status")
+        if "no such file" in result.stdout:
+            c.run("supervisord")
```

Comment:
same comment as in fabfile.py
if you shutdown supervisor, you are pretty sure that it is not started I think, so the `supervisorctl status` is for me not needed.